### ISSUE TITLE
How to save best trained model?

### ISSUE THREAD

**Issue by karndeepsingh** [unit8co/darts#527]
Hi,
I was training Transformer model on multiple time series problem. While doing model.fit() I have provided Training and test data for validation. Following things I want to know from here:

1. How is the training loss and validation loss is being calculated when I fit the model on data? What metrics is being used internally to show up the training loss and validation loss?
2. How to save best trained model which has very low validation as well as training loss?

3. When I use model.predict() after model.fit(), does this prediction happens with best trained weights or it happens with the latest trained weights that was updated during the last epoch?

Any guidance on these would be helpful.
Thanks

**Comment by dennisbader** [comment]
Hi @karndeepsingh and thanks for writing.

1. The training loss is evaluated between the predicted target and actual target values on the output chunk of every training data sample. The same is applied to evaluate the loss on the validation set.
The models use PyTorch criterions as loss functions. Per default this is [PyTorch's `MSELoss`](https://pytorch.org/docs/stable/generated/torch.nn.MSELoss.html) in most models.
You can use another PyTorch criterion by setting optional parameter `loss_fn=YourCriterion()` at model creation.
2. Per default, the model state is saved automatically including the last 5 epochs, and, if you supplied a validation set, the best overall model (with lowest validation loss). The save files can be found in your working directory under `./.darts/checkpoints/your_model_name/`
   - `your_model_name` per default will give a unique name. You can access it with `model.model_name`
   - you can define a custom model name `your_model_name` at model creation with optional parameter `model_name`
3. `model.predict()` uses the latest model state from the end of training (not necessarily the best).
    - You can load the best model with `model_best = model.load_from_checkpoint(model_name=model.model_name, best=True)`

**Comment by karndeepsingh** [comment]
@dennisbader  Thanks for answering.
I have been training on multiple time series and I am struggling to achieve better accuracy. I tried to even hyper tune these multi time-series using grid-search but grid search just takes a single series as an input to find the best hyperparameters for a particular series not for all Multiple series.

Pease help to find a better way to hyper tune the models on Multiple Time series. Right now, grid search just supports single time series hyperparameter tuning. 

Also, I need a advise to improve the performance of the model on Multi Series training. It take lot of time to train a model. Please provide some suggestion to improve the accuracy.

Thanks

**Comment by hrzn** [comment]
Hi @karndeepsingh yes current grid search works only on one series, but we have it on the roadmap to extend to multiple series. In the meantime I'd recommend that you start with small model sizes in order to iterate quicker, and maybe don't start with TransformerModel, which is big. (RegressionModel, RNNModel or NBEATSModel might be better to start with on multiple series). A GPU and casting your data to float32 can also help for some of the deep learning models.

**Comment by karndeepsingh** [comment]
> Hi @karndeepsingh yes current grid search works only on one series, but we have it on the roadmap to extend to multiple series. In the meantime I'd recommend that you start with small model sizes in order to iterate quicker, and maybe don't start with TransformerModel, which is big. (RegressionModel, RNNModel or NBEATSModel might be better to start with on multiple series). A GPU and casting your data to float32 can also help for some of the deep learning models.

@hrzn 
Thanks for answering. What is the difference between RNNModel and BLOCKRNNModel?

**Comment by pulkitvyas08** [comment]
Is there a change in the model saving path? I'm unable to find the saved model in `./.darts/checkpoints/your_model_name/`
